posts/2023/installing-farmos-on-ddev-with-composer/index.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Installing farmOS on DDEV with Composer
+date: 2023-05-17
+author: rgcarr
+slug: 2023/installing-farmos-on-ddev-with-composer
+---
+
+# DDEV
+
+For the recommended farmOS local development environment using Docker and Docker Compose, see the 
+[farmOS development environment docs](https://farmos.org/development/environment). That method works great in
+supported environments for developing farmOS core and small projects but can
+require additional configuration when working on larger projects in different
+environments like Colima. For these use-cases installing farmOS in a
+[DDEV](https://ddev.com/) environment may be more straightforward.
+
+> DDEV is an open source tool for running local web development environments for PHP, Python and Node.js, ready in minutes. Its powerful, flexible per-project environment configurations can be extended, version controlled, and shared. DDEV allows development teams to adopt a consistent Docker workflow without the complexities of bespoke configuration.
+
+DDEV's many features are what make it a great way to build Drupal and farmOS
+projects in a local environment:
+
+* Cross-platform, open source software.
+* Supports many versions of Docker or can be used with Colima.
+* When combined with Colima and Mutagen, performance is significantly improved.
+* Automatic HTTPS.
+* Run multiple projects concurrently.
+* Can be used to develop many other CMSs.
+* Support multiple databases (including MySQL, PostgreSQL, SQLite).
+* Lots of developer tools included, and is easy to extend and integrate.  
+
+# Installing farmOS with DDEV and Composer
+
+- First, follow the instructions to install and setup DDEV: https://ddev.readthedocs.io/en/latest/users/install/
+- Once installed, the following commands will create a DDEV project and install farmOS:
+
+```
+# Create a directory for the farmOS DDEV project
+mkdir farmos-project && cd farmos-project
+ddev config --project-type=drupal10 --docroot=web --create-docroot --database=postgres:15 --php-version=8.1
```

Comment:
````suggestion
# Installing farmOS with DDEV and Composer

- First, follow the instructions to install and setup DDEV: https://ddev.readthedocs.io/en/latest/users/install/
- Once installed, you can create a DDEV project and install farmOS. Your DDEV project is configured to use
a specific PHP and database version and can include additional DDEV addons and services. Here are two
examples of how to create and configure your DDEV project:

1. Use the `wotnak/ddev-farmos` addon to configure a DDEV project with the latest recommended versions
of PHP, PostgreSQL and PHP GEOS for farmOS. This addon will also configure the private filesystem and a Cron job for Drupal.
```
# Create a directory for the farmOS DDEV project
mkdir farmos-project && cd farmos-project
ddev config --project-type=php
ddev get wotnak/ddev-farmos
ddev composer create farmos/project:3.x-dev
ddev drush site:install farm --account-name=admin --account-pass=admin -y
ddev drush uli
```

2. If you prefer, you can also configure each aspect of the DDEV project yourself:
```
# Create a directory for the farmOS DDEV project
mkdir farmos-project && cd farmos-project
ddev config --project-type=drupal10 --docroot=web --create-docroot --database=postgres:15 --php-version=8.2
````

Reply:
@wotnak what do you think of this? I removed the `FARMOS_VERSION=2/3` variable complexity because very soon we will be recommending v3. Does that break anything?

Reply:
Looks good. Removing the `FARMOS_VERSION` variable doesn't break anything.